packages/headless/src/controllers/product-listing/range-facet/date-facet/headless-product-listing-date-facet.ts

```diff
@@ -0,0 +1,76 @@
+import {DateRangeRequest} from '../../../../features/facets/range-facets/date-facet-set/interfaces/request';
+import {DateFacetValue} from '../../../../features/facets/range-facets/date-facet-set/interfaces/response';
+
+import {getAnalyticsActionForToggleRangeFacetSelect} from '../../../../features/facets/range-facets/generic/range-facet-utils';
+import {DateFacetOptions} from '../../../core/facets/range-facet/date-facet/headless-date-facet-options';
+import {
+  buildCoreDateFacet,
+  buildDateRange,
+  DateFacet,
+  DateFacetProps,
+  DateFacetState,
+  DateRangeInput,
+  DateRangeOptions,
+} from '../../../core/facets/range-facet/date-facet/headless-core-date-facet';
+import {
+  logFacetClearAll,
+  logFacetUpdateSort,
+} from '../../../../features/facets/facet-set/facet-set-analytics-actions';
+import {RangeFacetSortCriterion} from '../../../../features/facets/range-facets/generic/interfaces/request';
+import {fetchProductListing} from '../../../../features/product-listing/product-listing-actions';
+import {ProductListingEngine} from '../../../../product-listing.index';
+
+export {
+  DateFacetOptions,
+  DateRangeInput,
+  DateRangeOptions,
+  DateRangeRequest,
+  buildDateRange,
+  DateFacetProps,
+  DateFacet,
+  DateFacetState,
+};
+
+/**
+ * Creates a `DateFacet` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `DateFacet` controller properties.
+ * @returns A `DateFacet` controller instance.
+ */
+export function buildDateFacet(
+  engine: ProductListingEngine,
+  props: DateFacetProps
+): DateFacet {
+  const coreController = buildCoreDateFacet(engine, props);
+  const dispatch = engine.dispatch;
+  const getFacetId = () => coreController.state.facetId;
+
+  return {
+    ...coreController,
+
+    deselectAll() {
+      coreController.deselectAll();
+      dispatch(fetchProductListing());
+      dispatch(logFacetClearAll(getFacetId()));
```

Comment:
Same comment regarding logging after a search.

Reply:
If we do that, I'd do a whole pass of all the components, because it's missing *everywhere* :smile: